toucan_connectors/one_drive/one_drive_connector.py

```diff
@@ -0,0 +1,116 @@
+import logging
+import os
+from pathlib import Path
+from typing import Optional
+
+import pandas as pd
+import requests
+from pydantic import Field
+
+from toucan_connectors.oauth2_connector.oauth2connector import (
+    OAuth2Connector,
+    OAuth2ConnectorConfig,
+)
+from toucan_connectors.toucan_connector import (
+    ConnectorSecretsForm,
+    ToucanConnector,
+    ToucanDataSource,
+)
+
+API_BASE_ROUTE: str = 'https://graph.microsoft.com/v1.0/me'
+AUTHORIZATION_URL: str = 'https://login.microsoftonline.com/common/oauth2/v2.0/authorize'
+TOKEN_URL: str = 'https://login.microsoftonline.com/common/oauth2/v2.0/token'
+
+
+class OneDriveDataSource(ToucanDataSource):
+    file: str
+    sheet: str
+    range: Optional[str]
+
+
+class OneDriveConnector(ToucanConnector):
+
+    data_source_model: OneDriveDataSource
+
+    _auth_flow = 'oauth2'
+    _oauth_trigger = 'instance'
+    oauth2_version = Field('1', **{'ui.hidden': True})
+    auth_flow_id: Optional[str]
+
+    scope: str = Field(
```

Comment:
Scope parameter is not return in this case and it is not possible to connect with oAuth2 flow

With my previous comment, I think the easyest solution is: 
`    _auth_flow = 'oauth2'
    _oauth_trigger = 'connector'
    oauth2_version = Field('1', **{'ui.hidden': True})
    auth_flow_id: Optional[str]

    client_id: str = Field(
        '',
        title='Client ID',
        description='The client id of you Snowflake integration',
        **{'ui.required': True},
    )
    client_secret: SecretStr = Field(
        '',
        title='Client Secret',
        description='The client secret of your Snowflake integration',
        **{'ui.required': True},
    )
    scope: str = Field(
        None,
        Title='Scope',
        description='The scope determines what type of access the app is granted when the user is signed in',
        placeholder='offline_access Files.Read',
    )`